### ISSUE TITLE
Keep comment and suggestion related events in changes page even though the corresponding string is removed

### ISSUE THREAD

**Issue by Geeyun-JY3** [WeblateOrg/weblate#7189]
### Describe the problem

After the string is removed, the related comments and suggestions would be also removed.
Then the comment and suggestion related events (Comment added, Suggestion accepted, Suggestion added, Suggestion removed, Suggestion removed during cleanup) are missing in changes page, while the comment and suggestion count wouldn’t change (reduce).

### Describe the solution you'd like

Keep comment and suggestion related events in changes page even though the corresponding string is removed.

**Comment by nijel** [comment]
That's true at every level - for example, when a translation is removed, all related changes are removed as well.

Why do you think it would be useful to keep these?

**Comment by Geeyun-JY3** [comment]
I think none of these events (including the events related to a translation as you mentioned) should be removed.
Because I think everything should be recorded in changes page at every level. Then users can learn what has happened at specific. And that is what changes page is intended for, right?

In addition, it may be a good supplement for some file format which would remove the old string and add a new one when a string changed (such as  gettext: https://github.com/WeblateOrg/weblate/discussions/7168#discussioncomment-2065875). 
Users can even know exactly what string is added or removed (Maybe this need some new kinds of event. Anyway, I would open another issue for it later)

**Comment by nijel** [comment]
The changes for removed string wouldn't be associated with new one. Because of that I think most users wouldn't find these.

**Comment by Geeyun-JY3** [comment]
Yeah. This is because the way some file formats handle the changed string. At least users can get some clues (maybe a search feature for changes needed?).

My starting point is changes page should record every event.

**Comment by nijel** [comment]
Currently, changes are kept as long the string still exists. Once the string is removed, related changes are removed as well. People typically look at string changes, or at recent changes (this can be at translation, component or project level). These would never see changes for deleted string (because in first case, the string is not there, in the second case, these would be somewhere far in the history).

Keeping additional changes in the database will increase its size, and that needs to be justified. Unfortunately, I still don't see the use-case for this.

**Comment by Geeyun-JY3** [comment]
Crowdin would record the comment and suggestion related events even if the string is removed. And this is another reason why I open this issue.
Maybe record everything is the use-case. Archive is important to me.

In addition, user page has count for user‘s work: [%(count)s translations](https://hosted.weblate.org/translate/weblate/application/en/?checksum=613469241a22a197), [%(count)s suggestions](https://hosted.weblate.org/translate/weblate/application/en/?checksum=c87c4b614af637b8), [%(count)s comments](https://hosted.weblate.org/translate/weblate/application/en/?checksum=4ec49137f6c3f7cb). 
Therefore, users should be able to retrieve their work which has a corresponding number of translations, comments and suggestions in some way. (Some users may know that they can get their translation work from TM, but it is not enough.)

Sometimes something related to the string has been well discussed before. Keeping these comments may help avoid duplicate comments and provide more related information for latter translators. So as the other related changes events.
Therefore, if it is really impossible for some file formats such as gettext to keep and assemble these comments and suggestions after the source string is changed, keeping these events makes it possible for users to find these comments.  Then a search feature for filtering events of changes can make it more convenient to quickly find the wanted comments (events).

**Comment by Geeyun-JY3** [comment]
This can be an offset for #7882
